client/rest-high-level/src/test/java/org/elasticsearch/client/RequestTests.java

```diff
@@ -249,6 +251,41 @@ public void testExists() {
         getAndExistsTest(Request::exists, HttpHead.METHOD_NAME);
     }
 
+    public void testIndicesExist() {
+        String[] indices = randomIndicesNames(1, 10);
+        boolean flatSettings = true;
+        boolean includeDefaults = true;
+        boolean local = true;
+        boolean ignoreUnavailable = false;
+        boolean allowNoIndices = true;
+        boolean expandToOpenIndices = false;
+        boolean expandToClosedIndices = false;
+
+        IndicesOptions indicesOptions = IndicesOptions.fromOptions(
+            ignoreUnavailable,
+            allowNoIndices,
+            expandToOpenIndices,
+            expandToClosedIndices
+        );
```

Comment:
could you please randomize these parameters like we do in other tests?

Reply:
That's what I planned doing after this. The first version with randomized parameters looked a bit ugly (because, when they are false, the request params are null, so the checks in the test are more complicated = more ugly).

Reply:
you need to checkout the [docs](https://github.com/elastic/docs) project. I run this command (from the root of the elasticsearch repo) to build only the java rest docs: `/path/to/elastic/docs/build_docs.pl --doc docs/java-rest/index.asciidoc --chunk 1 --out ~/temp/asciidoc  --open`

Reply:
Works like a charm...